apps/ejabberd/src/mongoose_api_admin.erl

```diff
@@ -88,26 +88,39 @@ rest_terminate(_Req, _State) ->
     ok.
 
 %% @doc Called for a method of type "DELETE"
-delete_resource(Req, #http_api_state{command_category = Category} = State) ->
-    [Command] = mongoose_commands:list(admin, Category, method_to_action(<<"DELETE">>)),
+delete_resource(Req, #http_api_state{command_category = Category, bindings = B} = State) ->
+    Arity = length(B),
+    Cmds = mongoose_commands:list(admin, Category, method_to_action(<<"DELETE">>)),
+    [Command] = [C || C <- Cmds, mongoose_commands:arity(C) == Arity],
     mongoose_api_common:process_request(<<"DELETE">>, Command, Req, State).
 
 %%--------------------------------------------------------------------
 %% Internal funs
 %%--------------------------------------------------------------------
 
 %% @doc Called for a method of type "GET"
-to_json(Req, #http_api_state{command_category = Category} = State) ->
-    [Command] = mongoose_commands:list(admin, Category, method_to_action(<<"GET">>)),
+to_json(Req, #http_api_state{command_category = Category, bindings = B} = State) ->
+    Cmds = mongoose_commands:list(admin, Category, method_to_action(<<"GET">>)),
+    Arity = length(B),
+    [Command] = [C || C <- Cmds, mongoose_commands:arity(C) == Arity],
     mongoose_api_common:process_request(<<"GET">>, Command, Req, State).
 
 
 %% @doc Called for a method of type "POST" and "PUT"
 from_json(Req, #http_api_state{command_category = Category,
-                               command_subcategory = SubCategory} = State) ->
+                               command_subcategory = SubCategory,
+                               bindings = B} = State) ->
     {Method, Req2} = cowboy_req:method(Req),
-    [Command] = mongoose_commands:list(admin, Category, method_to_action(Method), SubCategory),
-    mongoose_api_common:process_request(Method, Command, Req2, State).
+    Cmds = mongoose_commands:list(admin, Category, method_to_action(Method), SubCategory),
+    {ok, Body, _} = cowboy_req:body(Req),
+    {Data} = jiffy:decode(Body),
```

Comment:
parsing body is already done in `mongoose_api_common:process_request/4` few lines below.
Maybe it is possible to move this arity checks there[1] in case to avoid double parsing and code duplication?

[1] https://github.com/esl/MongooseIM/blob/master/apps/ejabberd/src/mongoose_api_common.erl#L101